### ISSUE TITLE
Empty series with index and "str" dtype is initialized with values "n" instead of NaN

### ISSUE THREAD

**Issue by samronsin** [pandas-dev/pandas#23838]
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
pd.Series(index=[0, 1 ,2], dtype="str")
```

#### Problem description

The code above yields 
```
0    n
1    n
2    n
dtype: object
```
in pandas 0.23 (values are the string 'n') but 
```
0    NaN
1    NaN
2    NaN
dtype: object
```
in pandas 0.22 as I would have expected.

#### Output of ``pd.show_versions()``
<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Darwin
OS-release: 18.2.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.4
pytest: None
pip: 18.1
setuptools: 40.6.2
Cython: None
numpy: 1.15.4
scipy: None
pyarrow: None
xarray: None
IPython: 7.1.1
sphinx: None
patsy: None
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

**Comment by jreback** [comment]
this was just closed by #22564 and is a duplicate of the issue

**Comment by samronsin** [comment]
OK thanks, I guess that's why did not find the issue !
